src/app.js

```diff
@@ -300,6 +300,24 @@ app.routeAsync("/groups/:groupName")
   .getAsync(endpoints.static.sendGatsbyEntrypoint)
 ;
 
+app.routeAsync("/groups/:groupName/members")
+  .getAsync(endpoints.groups.listMembers);
+
+app.routeAsync("/groups/:groupName/roles")
+  .getAsync(endpoints.groups.listRoles);
```

Comment:
This has me thinking about your remark in #547:

> If using the proposed routes, then they should come before more generic routes for datasets. Or, it might be preferable to change the routes instead to avoid conflict with user-provided paths, e.g. to `/groups/:groupName/-/logo` or similar.

With this PR + the proposed endpoints in that issue, we would have to document that `members`, `roles`, `overview`, and `logo` can't be dataset names, and there might come more down the road in the same realm of groups management.

I'd be in favor of adding an intermediate in the path so there would be only one restricted dataset name for all groups management endpoints. We'd just have to settle on the name.

Reply:
Nod. This sort of namespacing makes sense to me. I forwent it here following your lead in #570 and because I figured we had to document a few restrictions anyhow so it wouldn't be too hard to document a short list. But happy to namespace instead and make it easy to keep that list short. Regardless of what we do for the management endpoints, though, we also need to document that your dataset can't start with `narratives/`. ;-) 

Without namespacing, the list of invalid dataset names would be:

  - `narratives/**`
  - `members/**` (soon, [#581][])
  - `roles/**` (soon, [#581][])
  - `overview` (soon, [#570][])
  - `logo` (soon, [#570][])

[#570]: https://github.com/nextstrain/nextstrain.org/pull/570
[#581]: https://github.com/nextstrain/nextstrain.org/pull/581

With namespacing, it'd be just two items.

For the namespace itself, I'm fond of short characters unlikely to be in anyone's name, like:

```
/groups/:groupName/-/**
/groups/:groupName/_/**
/groups/:groupName/=/**
```

and there's precedent set by lots of other people's URLs for things like that, but something more mundane and self-descriptive might be better, e.g.:

```
/groups/:groupName/manage/**
```

Reply:
I haven't noticed short characters used by other people's URLs, though I've probably come across it before.

I'm a bit more inclined to the self-descriptive option since we'll need to code this into Nextstrain CLI. Short characters might be slightly easier on users but might not provide enough context to developers.

If the concern is clashing with user dataset names, another idea would be to combine the two (e.g. `/groups/:groupName/_manage/**`). Not sure if there's any precedent for that.

The word "manage" seems to imply some sort of edit access, so it might be confusing to users who can only read a list of members at `/groups/:groupName/manage/members` but not actually "manage" it. How about something more generic like "settings"?

Also, noting that dataset names can't be `narratives` either, and presumably we'd do the same for the new namespace here:

https://github.com/nextstrain/nextstrain.org/blob/e98a3ea1e573e693fe51fee7488fd8cf2a0c7a80/src/app.js#L303-L305

Reply:
Fair point about "manage". I guess I was thinking in terms of "management API", which doesn't imply a level of edit access to me.

"settings" would be fine by me and follows the convention of using a plural noun for a collection.

Reply:
Shall we go with "settings" here and in #570, or is this something worth discussing with a larger group since it's a decision that will stick?

For "settings" specifically, I'll shoot down my previous idea of `_settings` since:

1. I imagine users wouldn't want to name a dataset `settings` anyways, though I can't speak for them.
2. `narratives` is already an existing namespace without the `_` prefix, and it'd be good to keep these namespace names consistent.

Reply:
Agreed re: not going with `_settings`. If we _were_ going to go that route, I think I'd prefer `.X` (e.g. `.settings`) to keep with the filesystem-based convention for "hidden" or "private" or "system" things (whereas prefixing with `_` is a code-based convention for roughly the same).

I'd be comfortable making the call to use `settings` also happy to solicit input from a larger group by posting to #nextstrain-dev for a quick poll/thoughts.

Reply:
[Posted to #nextstrain-dev](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1661452428629519), but we can start working with it in our PRs now.